packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -212,6 +212,7 @@ export class GraphNG extends React.Component<GraphNGProps, GraphNGState> {
   }
 
   componentDidUpdate(prevProps: GraphNGProps) {
+    window.grafanaPerformanceMetrics?.add(PerformanceMetricName.GraphNGComponentUpdate, 1);
```

Comment:
@leeoniya made some changes in [the original PR](https://github.com/grafana/grafana/pull/42230) and this is the result - global object accessible directly from `grafana-ui` so we don't have to pass any callbacks from `grafana/core`

![image](https://user-images.githubusercontent.com/23451458/153406972-0d001dc9-4038-4668-947e-2e2d22a0f549.png)

I kind of like it! there is no performance hit as `window.grafanaPerformanceMetrics` is undefined unless we started collecting metrics via `grafanaRuntime.performanceMetrics.start()`

Reply:
I looked at https://github.com/jballant/webpack-strip-block and I don't like it to be honest - it will work for the basic use case such as the one in this PR, but it will be tricky in cases where we also use helper functions to calculate the metric value. Besides, these comments are noisyyyy


I think this is good enough for now - eventually we can add plugin which strips all `window.grafanaPerformanceMetrics?.add` statements as you suggested